pkg/simulation/sim-agent.go

```diff
@@ -0,0 +1,45 @@
+package simulation
+
+import (
+	"github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent1"
+	"github.com/SOMAS2021/SOMAS2021/pkg/agents/team1/agent2"
+	"github.com/SOMAS2021/SOMAS2021/pkg/agents/team6"
+	"github.com/SOMAS2021/SOMAS2021/pkg/infra"
+	"github.com/google/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func (sE *SimEnv) createNewAgent(tower *infra.Tower, i, floor int) {
+	// TODO: clean this looping, make a nice abs map
+	sE.Log("Creating new agent")
+	abs := []AgentNewFunc{agent1.New, agent2.New, team6.New}
+
+	uuid := uuid.New().String()
+	bagent, err := infra.NewBaseAgent(sE.world, i, sE.AgentHP, floor, uuid)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	custagent, err := abs[i](bagent)
```

Comment:
I'm not sure this is the best way of writing the code - I think it better for the custom agent's constructor to call the base agent constructor, rather than calling the base agent constructor directly here... but I'm happy for that to be fixed in a later PR. Let me know what you decide here and I can create an issue - I'm happy to just fix it myself when I have some free time since it's a not a critical issue.

Reply:
> I think it better for the custom agent's constructor to call the base agent constructor

afaik we avoided that because then the custom constructor can manipulate / break the base agent. Since base agent has the following properties used for state control:

```
id        string
hp        int
floor     int
inbox     *list.List
tower     *Tower
```

Exposing the constructor means exposing things like tower since we need to pass it to custom for a successful constructor.

Reply:
Yeah agreed with Jaafar here, happy to hear further thoughts though Jason @jzzheng22

Reply:
Low priority anyway - happy to keep this in for now but I'll probably think about it over the break